### PR TITLE
Enable .NET 6.0 unit tests for `dotnet test` commands

### PR DESCRIPTION
**Pull request by OlegRa**
These changes are related to the #158 issue opened recently:

- Added .NET 6.0 TFM as a new target for running unit tests
- All outdated dependent NuGet packages were upgraded
- Added missing `Micorsoft.NET.Test.Sdk` package
- Project files were cleaned up a little bit

**NB:* This comit most probably conflicts with commit 58411a082cb4e70e4979905ec803e956a7f304f8.

### CODE REVIEW

File: test/Lokad.ILPack.Tests/Lokad.ILPack.Tests.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp3.1;net5.0</TargetFrameworks>
+    <TargetFrameworks>netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
```
Comment (vermorel) [inline test/Lokad.ILPack.Tests/Lokad.ILPack.Tests.csproj:4]: Support for both `netcoreapp3.1` and `net5.0` ends this year https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core Please remove those as well.

**Review by vermorel — CHANGES_REQUESTED** [review]
Minor: keep only `net6.0`

**Comment by OlegRa** [comment]
> Minor: keep only `net6.0` for the test suites.

Fixed. The PR can be merged now.

**Comment by vermorel** [comment]
@OlegRa Perfect! I did manually rebase and squash.

**Pull request closed without merging**